graphql/util.go

```diff
@@ -235,6 +236,88 @@ func SchedulePatch(ctx context.Context, patchId string, version *model.Version,
 	}
 }
 
+type PatchProjectVariantsAndTasks struct {
+	Variants []BuildVariantUI
+	Tasks    []string
+}
+
+type BuildVariantUI struct {
+	Name        string
+	DisplayName string
+	Tasks       []string
+}
```

Comment:
I think we can use the PatchProject and ProjectBuildVariant structs instead of creating these two new structs. The only difference is that the Variants slice would contain pointers.

Reply:
returning an array of pointers is not ideal because interacting with that array of pointers involves looping through array and converting to values. this is especially true since `getPatchProjectVariantsAndTasks` will be generalized to be used between gql and REST service functions.

Reply:
both of these structs were removed and replaced with one that represents the data structure used in patch.go